File: ruby/bob/bob.rb
class Bob
  def self.hey(str)
    str.strip!
    return 'Fine. Be that way!' if str.empty?
    if str==str.upcase && str.match(/[A-Z]/)
      str[str.length-1]=='?' ? "Calm down, I know what I'm doing!" : 'Whoa, chill out!'
    else
      return 'Sure.' if str[str.length-1]=='?'
      'Whatever.'
    end
  end
end

module BookKeeping
  VERSION = 2
end

File: python/bob/bob.py
def hey(phrase):
    phrase = phrase.strip()
    if not phrase:
        return 'Fine. Be that way!'
    if phrase == phrase.upper() and phrase != phrase.lower():
        print('TRUE')
        if phrase[len(phrase)-1] == '?':
            return "Calm down, I know what I'm doing!"
        else:
            return 'Whoa, chill out!'
    if phrase[len(phrase)-1] == '?':
        return 'Sure.'
    return 'Whatever.'
